provider/agent_test.go

```diff
@@ -36,7 +36,7 @@ func TestAgent(t *testing.T) {
 					motd_file = "/etc/motd"
 					shutdown_script = "echo bye bye"
 					shutdown_script_timeout = 120
-					login_before_ready = false
+					startup_script_behavior = "blocking"
```

Comment:
nit: can we add some test cases when `startup_script_behavior` is non-blocking?

Reply:
Added a bunch of test cases, although since there's no logic now it's a bit silly. 😄